zebrad/tests/acceptance.rs

```diff
@@ -351,3 +354,69 @@ fn valid_generated_config(command: &str, expected_output: &str) -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn listener_address_test1() -> Result<()> {
+    listener_address(
+        "localhost:0",
+        Some("Opened Zcash protocol endpoint at 127.0.0.1:"),
+    )?;
+    Ok(())
+}
+#[test]
+fn listener_address_test2() -> Result<()> {
+    listener_address(
+        "127.0.0.1:0",
+        Some("Opened Zcash protocol endpoint at 127.0.0.1:"),
+    )?;
+    Ok(())
+}
+#[test]
+fn listener_address_test3() -> Result<()> {
+    listener_address(
+        "::1:0",
+        Some("Opened Zcash protocol endpoint at \\[::1\\]:"),
+    )?;
+    Ok(())
+}
+#[test]
+fn listener_address_test4() -> Result<()> {
+    listener_address("invalid", None)?;
+    Ok(())
+}
+
+fn listener_address(request: &str, expected_response: Option<&str>) -> Result<()> {
+    zebra_test::init();
+
+    // Create a temp dir to work in
+    let dir = TempDir::new("zebrad_tests")?;
+
+    // Create a default config file and add request as listen_addr
+    let (config_path, mut config) = default_config(dir.path().to_path_buf())?;
+    config.network.listen_addr = request.parse()?;
+    fs::File::create(&config_path)?.write_all(toml::to_string(&config)?.as_bytes())?;
+
+    // Start zebrad with created config and kill it at 5 seconds.
+    // This should be enough time to open port.
+    let mut child = get_child(
+        &["-c", config_path.to_str().unwrap(), "start"],
+        &dir.path().to_path_buf(),
+    )?;
+    std::thread::sleep(Duration::from_secs(5));
+    child.kill()?;
+
+    let output = child.wait_with_output()?;
+    let output = output.assert_failure()?;
+
+    // Check response
+    if expected_response.is_some() {
+        output.stdout_contains(expected_response.unwrap())?;
+        // Exit by kill
+        assert!(output.was_killed());
+    } else {
+        // Exit by panic
+        assert!(!output.was_killed());
+    }
```

Comment:
For invalid addresses, we should accept panics, or timeouts.

If there is a timeout, we should check that the listener line is *not* present in the output.